eventing/README.md

```diff
@@ -18,143 +18,171 @@ The above concerns are consistent with the
-  and presents them as CloudEvents.
-  ```bash
-  kubectl apply --filename https://storage.googleapis.com/knative-releases/eventing/latest/release-source-k8sevents.yaml
-  ```
-- [GitHub](https://github.com/knative/eventing/tree/master/pkg/sources/github)
-  collects pull request notifications and presents them as CloudEvents.
-  ```bash
-  kubectl apply --filename https://storage.googleapis.com/knative-releases/eventing/latest/release-source-github.yaml
-  ```
-- [GCP PubSub](https://github.com/knative/eventing/tree/master/pkg/sources/gcppubsub)
-  collects events published to a GCP PubSub topic and presents them as
-  CloudEvents.
-  ```bash
-  kubectl apply --filename https://storage.googleapis.com/knative-releases/eventing/latest/release-source-gcppubsub.yaml
-  ```
-
-### Flows
-
-Lastly, we have a higher-level abstraction called a Flow which bundles up the
-specification from the Source to the endpoint, optionally allowing you to choose
-the Channel and Bus which the event is routed over. (Otherwise, there is a
-default Bus used to provision a Channel.)
-
-- **Flow** is the top-level user-facing concept in Eventing; it describes the
-  desired path from an external Source of events to a destination that will
-  react to the events. There is only a single type of Flow, which is installed
-  by the core Knative Eventing install.
-
-## Supporting Libraries
-
-In addition to the above, there is a golang library at
-[`pkg/event`](https://github.com/knative/eventing/tree/master/pkg/event) for
-handling CloudEvents and easily creating a container which processes CloudEvents
-delivered over HTTP. You can look at several of the samples for usage examples.
+The eventing infrastructure supports two forms of event delivery at the moment:
+
+1. Direct delivery from a source to a single Service. In this case, the Source
+   is responsible for retrying or queueing events if the destination Service is
+   not available.
+2. Fan-out delivery from a source or Service response to multiple endpoints
+   using Channels and Subscriptions. In this case, the Channel implementation
+   ensures that messages are delivered to the requested destinations and should
+   buffer the events if the destination Service is unavailable.
+
+![Control plane object model](control-plane.png)
+
+The actual message forwarding is implemented by multiple data plane components
+which provide observability, persistence, and translation between different
+messaging protocols.
+
+![Data plane implementation](data-plane.png)
+
+<!-- TODO(evankanderson): add documentation for Kafka bus once it is available. -->
+
+## Sources
+
+Each source is a separate Kubernetes custom resource. This allows each type of
+Source to define the arguments and parameters needed to instantiate a source.
+Knative Eventing defines the following Sources in the
+`sources.eventing.knative.dev` API group. Types below are declared in golang
```

Comment:
I would like to be able to click on this spec and get to the object spec

Reply:
I don't think we have auto-generated nice documentation. The closest we have is godoc, which is not very friendly at the moment:

https://godoc.org/github.com/knative/eventing/pkg/apis/eventing/v1alpha1

I'll see if we can clean up some of the formatting in our godoc, but I'm not sure how we end up with kubernetes-style API documentation. Do you have a link to such a tool?

Reply:
Kubernetes API doc build details are here: https://kubernetes.io/docs/contribute/generate-ref-docs/kubernetes-api/